From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 4 Feb 2021 16:37:02 +0000
Subject: appDisplay: Do not bind popdown call to grabHelper onUngrab

grabHelper is passing a boolean argument to onUngrab() callback function
and since commit 1acbdcc9b3b0 we'd end up adding it to to the callback list or
we'd try to invoke it:

 (gnome-shell:3490851): Gjs-CRITICAL **: 17:19:20.460: JS ERROR:
  TypeError: func is not a function
  onComplete/<@/media/M2/GNOME/gnome-shell/js/ui/appDisplay.js:2407:56
  onComplete@/media/M2/GNOME/gnome-shell/js/ui/appDisplay.js:2407:40
  _makeEaseCallback/<@/media/M2/GNOME/gnome-shell/js/ui/environment.js:85:13
  _easeActor/<@/media/M2/GNOME/gnome-shell/js/ui/environment.js:170:64

Use an arrow function so that we can control the parameters we pass
to popdown.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1635>

(cherry picked from commit e641547ddf2c2afb42b4f2d3f250d84e1463a710)

Origin: upstream, 3.38.4, commit:9a95c8e997498e56352d054b3c648011daa1208f
---
 js/ui/appDisplay.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 004be08..399ab54 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -2489,8 +2489,10 @@ var AppFolderDialog = GObject.registerClass({
         if (this._isOpen)
             return;
 
-        this._isOpen = this._grabHelper.grab({ actor: this,
-                                               onUngrab: this.popdown.bind(this) });
+        this._isOpen = this._grabHelper.grab({
+            actor: this,
+            onUngrab: () => this.popdown(),
+        });
 
         if (!this._isOpen)
             return;
